testing/shared/src/main/scala/io/iteratee/testing/EqInstances.scala

```diff
@@ -9,8 +9,8 @@ import scala.Predef._
 trait EqInstances {
   implicit def eqThrowable: Eq[Throwable] = Eq.fromUniversalEquals
 
-  implicit def eqEnumerator[F[_]: Monad, A: Eq](implicit eq: Eq[F[Vector[A]]]): Eq[Enumerator[F, A]] =
-    eq.on[Enumerator[F, A]](_.toVector)
+  implicit def eqEnumerator[F[_]: Monad, A: Eq](implicit eqFVA: Eq[F[Vector[A]]]): Eq[Enumerator[F, A]] =
+    Eq.by[Enumerator[F, A], F[Vector[A]]](_.toVector)(eqFVA)
```

Comment:
scalac does good job with inference here
```scala
Eq.by(_.toVector)
```
is sufficient

Reply:
Oh, thanks—I'd tried that but with the explicit `eqFVA`, which doesn't work. I guess there's no reason to pass the `F[Vector[A]]` instance explicitly, though.